app/models/miq_ae_domain.rb

```diff
@@ -100,6 +100,17 @@ def editable_property?(property)
 
   alias editable_contents? editable?
 
+  def update_attributes(hash)
+    not_editable = hash.keys.reject { |p| editable_property?(p) }
+    if not_editable.present?
+      msg = "Not editable attributes [#{not_editable.join(",")}] in [#{hash}] for domain update"
+      $miq_ae_logger.error(msg)
+      raise msg
```

Comment:
@mkanoor Is there an existing Automate error type we should use here or go with `ActiveRecord::RecordInvalid` which the `update!` method uses.

Reply:
@gmcculloug @lfu We don't have any exceptions for invalid attributes, the ActiveRecord::RecordInvalid is a good option to start with.